test/test_cases.cpp

```diff
@@ -0,0 +1,1279 @@
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(42, uut_pod[0]);
+  EXPECT_EQ(42, uut_pod[1]);
+  EXPECT_EQ(42, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("42", uut_obj[0]);
+  EXPECT_EQ("42", uut_obj[1]);
+  EXPECT_EQ("42", uut_obj[2]);
+}
+
+TEST_F(AssignMethodTests, AssignWithIteratorsLargerThanOriginal) {
+  std::vector<int32_t> std_vector_source_pod(5, 42);
+  std::vector<std::string> std_vector_source_obj(5, "42");
+
+  ekuvector<int32_t> uut_pod({97, 98, 99});
+  ekuvector<std::string> uut_obj({"97", "98", "99"});
+
+  uut_pod.assign(std_vector_source_pod.begin(), std_vector_source_pod.end());
+  uut_obj.assign(std_vector_source_obj.begin(), std_vector_source_obj.end());
+
+  ASSERT_EQ(5, uut_pod.size());
+  EXPECT_EQ(42, uut_pod[0]);
+  EXPECT_EQ(42, uut_pod[1]);
+  EXPECT_EQ(42, uut_pod[2]);
+  EXPECT_EQ(42, uut_pod[3]);
+  EXPECT_EQ(42, uut_pod[4]);
+
+  ASSERT_EQ(5, uut_obj.size());
+  EXPECT_EQ("42", uut_obj[0]);
+  EXPECT_EQ("42", uut_obj[1]);
+  EXPECT_EQ("42", uut_obj[2]);
+  EXPECT_EQ("42", uut_obj[3]);
+  EXPECT_EQ("42", uut_obj[4]);
+}
+
+TEST_F(AssignMethodTests, AssignWithInitListSmallerThanOriginal) {
+  std::initializer_list<int32_t> std_init_list_source_pod({42, 42, 42});
+  std::initializer_list<std::string> std_init_list_source_obj(
+      {"42", "42", "42"});
+
+  ekuvector<int32_t> uut_pod({97, 98, 99});
+  ekuvector<std::string> uut_obj({"97", "98", "99"});
+
+  uut_pod.assign(std_init_list_source_pod);
+  uut_obj.assign(std_init_list_source_obj);
+
+  ASSERT_EQ(3, uut_pod.size());
+  EXPECT_EQ(42, uut_pod[0]);
+  EXPECT_EQ(42, uut_pod[1]);
+  EXPECT_EQ(42, uut_pod[2]);
+
+  ASSERT_EQ(3, uut_obj.size());
+  EXPECT_EQ("42", uut_obj[0]);
+  EXPECT_EQ("42", uut_obj[1]);
+  EXPECT_EQ("42", uut_obj[2]);
+}
+
+TEST_F(AssignMethodTests, AssignWithInitListLargerThanOriginal) {
+  std::initializer_list<int32_t> std_init_list_source_pod({42, 42, 42, 42, 42});
+  std::initializer_list<std::string> std_init_list_source_obj(
+      {"42", "42", "42", "42", "42"});
+
+  ekuvector<int32_t> uut_pod({97, 98, 99});
+  ekuvector<std::string> uut_obj({"97", "98", "99"});
+
+  uut_pod.assign(std_init_list_source_pod);
+  uut_obj.assign(std_init_list_source_obj);
+
+  ASSERT_EQ(5, uut_pod.size());
+  EXPECT_EQ(42, uut_pod[0]);
+  EXPECT_EQ(42, uut_pod[1]);
+  EXPECT_EQ(42, uut_pod[2]);
+  EXPECT_EQ(42, uut_pod[3]);
+  EXPECT_EQ(42, uut_pod[4]);
+
+  ASSERT_EQ(5, uut_obj.size());
+  EXPECT_EQ("42", uut_obj[0]);
+  EXPECT_EQ("42", uut_obj[1]);
+  EXPECT_EQ("42", uut_obj[2]);
+  EXPECT_EQ("42", uut_obj[3]);
+  EXPECT_EQ("42", uut_obj[4]);
+}
+
+class ElementAccessTests : public EkuVectorTests {};
+
+TEST_F(ElementAccessTests, AtMemberNonConst) {
+  ekuvector<int32_t> uut_pod({97, 98, 99});
+  ekuvector<std::string> uut_obj({"97", "98", "99"});
+
+  EXPECT_EQ(97, uut_pod.at(0));
+  EXPECT_EQ(98, uut_pod.at(1));
+  EXPECT_EQ(99, uut_pod.at(2));
+  EXPECT_THROW(uut_pod.at(3), std::out_of_range);
```

Comment:
Would you add a negative index?

Reply:
[Done](https://github.com/glpuga/ekucontainers/pull/1/commits/e49547511c284c568f53151d326117c2bb28e24f).

Reply:
Thanks!